ccm_web/server/src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,39 @@
+import { Request } from 'express'
+import { Strategy as PassportJwtStrategy } from 'passport-jwt'
+import { Injectable } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import { PassportStrategy } from '@nestjs/passport'
+
+import { JwtPayload } from './auth.interfaces'
+import { UserNotFoundError } from '../user/user.errors'
+import { User } from '../user/user.model'
+import { UserService } from '../user/user.service'
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(PassportJwtStrategy) {
+  constructor (
+    private readonly configService: ConfigService,
+    private readonly userService: UserService
+  ) {
+    super({
+      jwtFromRequest: (req: Request): string | null => {
+        if (req.signedCookies.jwt !== undefined) {
+          const token = req.signedCookies.jwt as string
+          return token
```

Comment:
I looked some of the Req/Res for Course API calls from the Browser console, and I don't see that you are passing this JWT token as bearer token with request from client, but more attached this as a Cookie to requests. By reading through the [documentation](https://docs.nestjs.com/security/authentication) from nestJS they shared bearer approach, there is no right or wrong here. But what was you thinking here, if I understood  this correctly 

here is some context from what I mean from documentation....
```
$ # GET /profile
$ curl http://localhost:3000/profile
$ # result -> {"statusCode":401,"error":"Unauthorized"}

$ # POST /auth/login
$ curl -X POST http://localhost:3000/auth/login -d '{"username": "john", "password": "changeme"}' -H "Content-Type: application/json"
$ # result -> {"access_token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2Vybm... }

$ # GET /profile using access_token returned from previous step as bearer code
$ curl http://localhost:3000/profile -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2Vybm..."
$ # result -> {"userId":1,"username":"john"}
```

Reply:
Based on the research I did, I was operating with the understanding that it was more secure to store the JWT in a cookie than any other way, but that it did open you up to CSRF attacks -- hence the need for the `csurf` solution. I open to changing this, but we need to be careful regardless to ensure we're covering our bases. @jonespm, any thoughts?

Here's a couple places I read about the preference for storing JWTs in cookies (including some I just found that echo similar sentiments):
- https://dev.to/gkoniaris/how-to-securely-store-jwt-tokens-51cf
- https://medium.com/@ryanchenkie_40935/react-authentication-how-to-store-jwt-in-a-cookie-346519310e81
- https://stormpath.com/blog/where-to-store-your-jwts-cookies-vs-html5-web-storage
- https://indepth.dev/posts/1382/localstorage-vs-cookies